python/python/lance/ray/sink.py

```diff
@@ -150,6 +150,33 @@ def on_write_complete(
         self,
         write_results: List[List[Tuple[str, str]]],
     ):
+        import warnings
+
+        if not write_results:
+            warnings.warn(
+                "write_results is empty.",
+                DeprecationWarning,
+            )
+            return "Empty list"
+        if (
+            not isinstance(write_results, list)
+            or not isinstance(write_results[0], list)
+        ) and not hasattr(write_results, "write_returns"):
+            warnings.warn(
+                "write_results type is wrong. please check version",
```

Comment:
```suggestion
                "ray versions >= 2.38 and < 3.0 are unable to write Lance datasets, please upgrade your ray version",
```

Can we be more specific with the error message?